src/hooks/useTheming.js

```diff
@@ -0,0 +1,92 @@
+// eslint-disable-next-line no-unused-vars
+import React, { useState, useEffect } from 'react';
+
+/**
+ * @typedef {'light'|'dark'} ThemeValues
+ */
+
+/**
+ * A hooks to customize the app theming.
+ * The app theme is customized using CSS variables inside a [data-theme='dark']
+ * block, e.g.:
+ *
+ * ```css
+ *  :root {
+ *    --shadow-color: blue;
+ *  }
+ *
+ *  [data-theme="dark"] {
+ *    --shadow-color: red;
+ *  }
+ * ```
+ *
+ * @returns {[ThemeValues, setTheme]} - The current color scheme and a function to
+ *                                to update it.
+ */
+const useTheming = () => {
+  const [theme, setTheme] = useState('light');
+
+  /**
+   * Checks if the system already supports the
+   * prefers-color-scheme media query. If so, returns
+   * the query itself.
+   * Returns null if the system doesn't support it yet.
+   *
+   * @returns {(ThemeValues|null)} - The system theme.
+   */
+  const getSystemTheme = () => {
+    // If the browser doesn't support CSS media queries,
+    // returns false.
+    if (!window.matchMedia) return null;
+
+    if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
+      return 'dark';
+    }
+
+    return 'light';
+  };
+
+  /**
+   * Checks if a theme is saved on the local storage.
+   * This function is used as a fallback if the system theme
+   * isn't available.
+   *
+   * @returns {(ThemeValues)} - The theme stored in the local storage.
+   */
+  const getLocalStorageTheme = () => {
```

Comment:
Could this be used as the initial check for a defined theme? That is, check for a theme already set in localStorage and fall-back to `getSystemTheme()` if it's undefined. That way we could provide a toggle switch if someone doesn't want to use their current system theme for whatever reason.

Reply:
I changed the function and now it's first checking if there's a theme already set on the local storage. If not, it fallbacks to the system color scheme.